### PR TITLE
chore(Styleguide): Add prop descriptions Gamut A-F GM-42

### PR DESCRIPTION
**Pull request by codecaaron**
## Changes
- Adds prop descriptions for docgen typescript for prop tables on components A-F.
- Moves card to atoms

### Further work
- Button `fitText` prop no longer is mapped to a style and is no op.  (New ticket: GM-52)
- `ContentContainer`, `FlexGrid`, and others do not have the standardized `as` prop for specifying a custom element.  (New ticket: GM-53)

### PR Checklist
- [ ] Related to Abstract designs:
- [x] Related to JIRA ticket: GM-42
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0cecb9e45bb4411642129b--codecademy-styleguide.netlify.app>https://5f0cecb9e45bb4411642129b--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0cecb9e45bb4411642129b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.40-alpha.0ffecc.0<br/>@codecademy/gamut@9.4.1-alpha.0ffecc.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.41-alpha.dc585d.0<br/>@codecademy/gamut@9.5.1-alpha.dc585d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0dc5369d3f07ee03e037fa--codecademy-styleguide.netlify.app>https://5f0dc5369d3f07ee03e037fa--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0dc5369d3f07ee03e037fa>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.41-alpha.23b189.0<br/>@codecademy/gamut@9.5.1-alpha.23b189.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0dd74b0b196379ec6a4369--codecademy-styleguide.netlify.app>https://5f0dd74b0b196379ec6a4369--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0dd74b0b196379ec6a4369>Deploy Logs</a></p>

File: packages/gamut/src/AccordionButton/index.tsx [hunk 1]
```diff
@@ -8,9 +8,21 @@ import ButtonBase from '../ButtonBase';
 
 export type AccordionButtonProps = {
   className?: string;
+  /**
+   * Whether or not the button should display as open or closed.
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/AccordionButton/index.tsx:12]: Nit: _or not_ is filler.

```suggestion
   * Whether the button should display as open or closed.
```

File: packages/gamut/src/AccordionButton/index.tsx [hunk 2]
```diff
@@ -8,9 +8,21 @@ import ButtonBase from '../ButtonBase';
...
+   */
   expanded?: boolean;
+  /**
+   * Called when the button is clicked.
+   */
   onClick: () => void;
+  /**
+   * Variant that determines the size of the button.
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/AccordionButton/index.tsx:20]:
```suggestion
   * Determines the size of the button.
```

File: packages/gamut/src/AppBar/AppBarSection.tsx [hunk 3]
```diff
@@ -3,6 +3,9 @@ import cx from 'classnames';
 import styles from './styles/index.module.scss';
 
 export type AppBarSectionProps = {
+  /**
+   * Position of the the section within the AppBar
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/AppBar/AppBarSection.tsx:7]:
```suggestion
   * Position of the the section within the AppBar.
```

File: packages/gamut/src/AppBar/index.tsx [hunk 4]
```diff
@@ -5,6 +5,9 @@ import styles from './styles/index.module.scss';
 
 export type AppBarProps = {
   className?: string;
+  /**
+   * Whether the container of should be larger than the default content size.
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/AppBar/index.tsx:9]: "of" - ?

File: packages/gamut/src/Button/index.tsx [hunk 5]
```diff
@@ -42,25 +42,71 @@ export type ButtonProps = HTMLAttributes<HTMLButtonElement> & {
...
+  /**
+   * Variant that controls the background and text color of the button
+   * */
   theme?: string;
   type?: string;
+  /**
+   * Variant that underlines the text of the button.
+   */
   underline?: boolean;
+  /** DEPRECATED: This is a no-op prop */
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Button/index.tsx:108]: Technically the `@deprecated` tag is what you're looking for.

```suggestion
  /** @deprecated This is a no-op prop */
```

File: packages/gamut/src/FlexGrid/Col.tsx [hunk 6]
```diff
@@ -54,19 +54,28 @@ function getClassNames(props: ColProps) {
 }
 
 export type ColSizing = {
+  /** Size of the column on screensizes greater than `xs` */
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/FlexGrid/Col.tsx:57]: Was this intentional?

```suggestion
  /** Size of the column on screen sizes greater than `xs` */
```

**Comment by codecaaron** [comment]
> This is fantastic, thanks!
> 
> I left a few nits early on that are repeated later in props, but up to you if you want to apply them. Just some copy editing 📝

Thank you!  The boolean descriptions always throw me off.  Found some other places where I was doing the screensizes thing.  Great suggestions :D

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.41-alpha.807f4a.0<br/>@codecademy/gamut@9.5.1-alpha.807f4a.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.41-alpha.48511e.0<br/>@codecademy/gamut@9.5.1-alpha.48511e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0e0cb0fd40063d87ef0c2b--codecademy-styleguide.netlify.app>https://5f0e0cb0fd40063d87ef0c2b--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0e0cb0fd40063d87ef0c2b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0e0ef13f36956f07a8cc38--codecademy-styleguide.netlify.app>https://5f0e0ef13f36956f07a8cc38--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0e0ef13f36956f07a8cc38>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.41-alpha.9037d8.0<br/>@codecademy/gamut@9.5.1-alpha.9037d8.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.41-alpha.4917cf.0<br/>@codecademy/gamut@9.5.1-alpha.4917cf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f0e21931e3ece02875cf0c3--codecademy-styleguide.netlify.app>https://5f0e21931e3ece02875cf0c3--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f0e21931e3ece02875cf0c3>Deploy Logs</a></p>

**Pull request merged**